libmscore/barline.cpp

```diff
@@ -636,8 +676,21 @@ Element* BarLine::drop(EditData& data)
                   else
                         undoChangeBarLineType(this, st, false);
                   }
-            else
+            else {
+                  //---------------------------------------------
+                  //    Update repeat flags
+                  //---------------------------------------------
+
+                  Measure* m  = segment()->measure();
+                  // drop on a end-start-repeat barline remove also the start repeat on next measure
```

Comment:
Should it, though?  This was pointed out in the forum.

Reply:
IIRC that is code taken from before this feature got removed.